dashboard/app/models/levels/dsl_defined.rb

```diff
@@ -120,6 +125,7 @@ def self.decrypt_dsl_text_if_necessary(dsl_text)
   end
 
   def clone_with_name(new_name)
+    raise "A level named '#{new_name}' already exists" if Level.find_by_name(new_name)
```

Comment:
Can't you just use the existing level instead of throwing an exception?

Reply:
Good observation. We could do this, however I'm not sure this makes sense to change, because it currently matches the contract for Level.clone_with_name, which I've now clarified in level.rb.

If we are cloning a single level via clone_with_name, then the caller (via rails console) should not need to even call this method if the new level already exists. When cloning an entire script, clone_with_suffix will take care of checking whether the level exists already and using that instead, so again there is no advantage to finding the existing level here.

Incidentally, this reminded me that we are pretty much always going to want `allow_existing` to be true, so I've gone ahead and removed it as a param from all places.